README.rst

```diff
@@ -73,42 +70,18 @@ Dependencies
 
 The minimum required dependencies to run MNE-Python are:
 
+.. ↓↓↓ BEGIN CORE DEPS LIST. DO NOT EDIT! HANDLED BY PRE-COMMIT HOOK ↓↓↓
 - `Python <https://www.python.org>`__ ≥ 3.9
-- `NumPy <https://numpy.org>`__ ≥ 1.24
-- `SciPy <https://scipy.org>`__ ≥ 1.10
+- `NumPy <https://numpy.org>`__ ≥ 1.23, < 3
```

Comment:
Hah this is more of a `dev`-related thing than a user-facing one, we should probably remove this (just for NumPy for now)